### PR TITLE
Build support custom resource

### PR DESCRIPTION
**Pull request by facethesunsea**
### What this PR does / Why we need it:

- build step support custom resource

image

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

### Check List 

- [ ] Docs have been added / updated
- [ ] Unit test / Integration test for the changes have been added
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

## More information

### CODE REVIEW

File: src/components/projects/build/build_env.vue [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+            </el-option>
+          </el-select>
+        </el-form-item>
+      </el-col>
+      <el-col :span="mini ? 12 : 16">
+        <el-form-item :label="mini ? '资源' : '资源规格'" label-width="inherit" required>
+          <el-select size="small" v-model="pre_build.res_req" placeholder="请选择">
+            <el-option label="高 | CPU: 16 核 内存: 32 GB" value="high"></el-option>
+            <el-option label="中 | CPU: 8 核 内存: 16 GB" value="medium"></el-option>
+            <el-option label="低 | CPU: 4 核 内存: 8 GB" value="low"></el-option>
```
Comment (leozhang2018) [inline src/components/projects/build/build_env.vue:26]: 另外两项应该使用单独的 el-col 处理。
以适配不同宽度。
image
image

File: src/components/projects/build/build_env.vue [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    }
+
+    this.validateMemoryLimit = (rule, value, callback) => {
+      if (!value) {
+        callback(new Error('请输入自定义内存'))
+        return
+      }
+      const exec = /^(\d+)Mi$/.exec(value)
+      if (!exec) {
+        callback(new Error('请输入正确内存格式，例如：512Mi'))
```
Comment (leozhang2018) [inline src/components/projects/build/build_env.vue:101]: 中英文混排
512 Mi

File: src/components/projects/build/build_env.vue [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+      } else if (exec[1] <= 0) {
+        callback(new Error('内存必须大于0'))
```
Comment (leozhang2018) [inline src/components/projects/build/build_env.vue:103]: 中英文混排
内存必须大于 0

File: src/components/projects/build/build_env.vue [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+              :prop="`${propPre}.res_req_spec.cpu_limit`"
+              :rules="{ validator: validateCpuLimit, trigger: ['change', 'blur'] }"
+            >
+              <el-input v-model="pre_build.res_req_spec.cpu_limit" placeholder="自定义 CPU" size="small"></el-input>
+            </el-form-item>
+
+            <el-form-item
+              label="内存(Mi)"
+              label-width="70px"
+              :prop="`${propPre}.res_req_spec.memory_limit`"
```
Comment (leozhang2018) [inline src/components/projects/build/build_env.vue:44]: image

无需填写单位

**Review by leozhang2018 — CHANGES_REQUESTED** [review]

File: src/components/projects/build/build_env.vue (hunk 1 above)
Comment (facethesunsea) [reply src/components/projects/build/build_env.vue:26]: image
这种 el-col 不合适

**Review by leozhang2018 — APPROVED** [review]
LGTM

**Pull request merged**